src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -13,6 +13,7 @@ private struct DateTimeParseData
             public int Year;
             public int Month;
             public int Day;
+            public bool IsCalendarDateOnly;
```

Comment:
I don't know if it would help with https://github.com/dotnet/runtime/issues/69447#issuecomment-1161121510, but defining this field here will increase the size of this struct from 40 bytes to 48 bytes on 64-bit.  If you instead move this field to the end, the size should stay at 40.